multi-tenant-configuration/configure_users.py

```diff
@@ -0,0 +1,378 @@
+    if not existing_user:
+        # create user if it does not exist on tenant (Ask for permission)
+        action_allowed = check_or_ask_for_permission(
+            target_type='user',
+            action='create',
+            target_name=user['username'],
+            tenant_id=tenant_id
+        )
+        if action_allowed:
+            create_user(account=user, tenant_id=tenant_id)
+    else:
+        # Check if password is correct and if the account has External API access.
+        __check_api_access(user=user, tenant_id=tenant_id)
+        # Check if the user roles match the roles in the configuration file.
+        __check_user_roles(user, existing_user, tenant_id)
+        # check for unexpected roles in the effective roles.
+        __check_effective_user_roles(user, tenant_id)
+
+
+def __check_api_access(user: dict, tenant_id: str):
+    """
+    Checks if the user defined in the config has access to the API.
+    The check tries to login with the username and password defined in the config,
+    and sends a get request to '/api/info/me' .
+    If check fails, asks for user permission to update user.
+    :param user: The user defined in the config
+    :type user: Dict
+    :param tenant_id: The target tenant
+    :type tenant_id: String
+    """
+    log(f"Checking API access for user {user['username']}")
+
+    url = f'{CONFIG.tenant_urls[tenant_id]}/api/info/me'
+    headers = {}
+    login = {
+        'user': user['username'],
+        'password': user['password']
+    }
+    try:
+        get_request(url, login, '/api/info/me', headers=headers, use_digest=False)
+    except RequestError:
+        print(f"User {user['username']} has no API Access")
+        action_allowed = check_or_ask_for_permission(
+            target_type='user',
+            action='configure user',
+            target_name=user['username'],
+            tenant_id=tenant_id
+        )
+        if action_allowed:
+            update_user(tenant_id, user=user)
+    except Exception as e:
+        print('ERROR: Failed to check for API access.')
+        print(str(e))
+
+
+def __check_effective_user_roles(user: dict, tenant_id: str):
+    """
+    Checks if the effective user roles of the user contain unexpected roles.
+    prints a warning for each unexpected role.
+    :param user: User containing the username for whom to retrieve the roles
+    :type user: Dict
+    :param tenant_id: The ID of the target tenant
+    :type tenant_id: String
+    """
+    log(f"Check effective user roles of user {user['username']}")
+
+    effective_user_roles = get_user_roles(user['username'], tenant_id)
+    for role in effective_user_roles:
+        for unexpected_role in UNEXPECTED_ROLES:
+            if unexpected_role in role:
+                print(f"WARNING: Unexpected role found for User {user['username']}: {role}")
+
+
+def __check_user_roles(user: dict, existing_user: dict, tenant_id: str):
```

Comment:
That's pretty much the same that's happening for group roles, no? And the group member check is also similar. There might be some way of abstracting this away, but I'm not sure yet if the resulting complexity would be wort it.